### ISSUE TITLE
Inherited class slots are not reflecting the correct range in the python and markdown files

### ISSUE THREAD

**Issue by wdduncan** [linkml/linkml#497]
**Describe the bug**
@cmungall found a case in the  [NMDC Schema](https://microbiomedata.github.io/nmdc-schema/) in which the a slot inherited a slot, but the inherited slot does not have the correct range.

**To Reproduce**
The class [biosample process](https://microbiomedata.github.io/nmdc-schema/BiosampleProcessing/) species the range of `has input` as `biosample`.  
The class [omics processing](https://microbiomedata.github.io/nmdc-schema/OmicsProcessing/#class-omics-processing) extends `biosample processing`, but the range of its `has input` slot is `named thing`.   
This is also reflected in the generated python file [nmdc.py](https://github.com/microbiomedata/nmdc-schema/blob/8a528782313d667ff2261bcd61197478a6bdab40/python/nmdc.py#L1257).
```
has_input: Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]] = empty_list()
```

**Comment by sierra-moxon** [comment]
Hi @pkalita-lbl, @sujaypatil96  - could you test if this is still a problem for NMDC schema and close this issue if not?  Hopefully it is an easy test :)

**Comment by pkalita-lbl** [comment]
It looks like the original issue is about both generated documentation and generated Python classes. Currently, the documentation appears to be correct but the issue in the generated Python class still exists. I think I can make a simpler example outside of the NMDC schema. I will update this case when I do.

**Comment by vincentvialard** [comment]
> Then you get a `Delta` class that declares both `s1` and `s2` and the type of `s2` is incorrectly listed as `Alpha`:

I suppose you meant:

> ... and the type of **`s1`** is incorrectly listed as `Alpha`:

as `s1` should be `Beta` as specified for instances of Gamma.

**Comment by pkalita-lbl** [comment]
Yes, thank you for pointing that out. I have edited my previous comment to avoid confusion.
